File: Grid-Challenge.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'gridChallenge' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING_ARRAY grid as parameter.
     */

    public static string gridChallenge(List<string> grid)
    {
        int rows = grid.Count;
        int cols = grid[0].Length;

        for (int i = 0; i < rows; i++)
        {
            char[] chars = grid[i].ToCharArray();
            Array.Sort(chars);
            grid[i] = new string(chars);
        }

        for (int col = 0; col < cols; col++)
        {
            char prev = grid[0][col];

            for (int row = 1; row < rows; row++)
            {
                char current = grid[row][col];
                if (current < prev)
                    return "NO";

                prev = current;
            }
        }

        return "YES";
    }    
}

File: Grid-Challenge.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'gridChallenge' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING_ARRAY grid as parameter.
     */

    public static String gridChallenge(List<String> grid) {
        int rows = grid.size();
        int cols = grid.get(0).length();

        for (int i = 0; i < rows; i++) {
            char[] chars = grid.get(i).toCharArray();
            Arrays.sort(chars);
            grid.set(i, new String(chars));
        }

        for (int col = 0; col < cols; col++) {
            char prev = grid.get(0).charAt(col);

            for (int row = 1; row < rows; row++) {
                char current = grid.get(row).charAt(col);
                if (current < prev)
                    return "NO";

                prev = current;
            }
        }

        return "YES";
    }
}

